rust/connlib/tunnel/src/io.rs

```diff
@@ -82,13 +83,54 @@ impl Default for Buffers {
     }
 }
 
-pub enum Input<D, I> {
-    Timeout(Instant),
-    Device(D),
-    Network(I),
-    TcpDnsQuery(l4_tcp_dns_server::Query),
-    UdpDnsQuery(l4_udp_dns_server::Query),
-    DnsResponse(dns::RecursiveResponse),
+pub struct Input<D, I> {
```

Comment:
nit: some comments perhaps? 

```suggestion
/// Represents all IO sources that may be ready during a single event-loop tick.
///
/// This structure allows us to batch-process multiple ready sources rather than
/// handling them one at a time, improving fairness and preventing starvation.
pub struct Input<D, I> {
```